compiler/ast/ast_query.nim

```diff
@@ -557,7 +557,15 @@ when false:
       if n[i].containsNil: return true
 
 
-template hasDestructor*(t: PType): bool = tfHasAsgn in t.flags
+func hasDestructor*(t: PType): bool {.inline.} =
```

Comment:
Separate from this PR/Tangent: I think it might be a good idea to think about moving queries that are only true after some amount of analysis to go into dedicated modules.